File: .temp/translate_names.mjs
/**
 * 批量翻译游戏名称
 * 使用 Google Translate 免费接口
 */

import fs from 'fs';
import https from 'https';

const INPUT_FILE = '../data/games_to_translate.json';
const OUTPUT_FILE = '../data/games_translated.json';

function translate(text) {
  return new Promise((resolve) => {
    const url = `https://translate.googleapis.com/translate_a/single?client=gtx&sl=en&tl=zh-CN&dt=t&q=${encodeURIComponent(text)}`;

    https.get(url, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try {
          const result = JSON.parse(data);
          if (result && result[0] && result[0][0]) {
            resolve(result[0][0][0]);
          } else {
            resolve(text);
          }
        } catch (e) {
          resolve(text);
        }
      });
    }).on('error', () => resolve(text));
  });
}

async function main() {
  const games = JSON.parse(fs.readFileSync(INPUT_FILE, 'utf-8'));
  console.log(`需要翻译的游戏数量: ${games.length}`);

  const results = [];
  const batchSize = 10;

  for (let i = 0; i < games.length; i += batchSize) {
    const batch = games.slice(i, i + batchSize);
    console.log(`[进度] ${i}/${games.length}...`);

    const promises = batch.map(async (game) => {
      const translated = await translate(game.nameEn);
      return {
        appId: game.appId,
        nameEn: game.nameEn,
        nameZh: translated
      };
    });

    const batchResults = await Promise.all(promises);
    results.push(...batchResults);

    // 保存中间结果
    fs.writeFileSync(OUTPUT_FILE, JSON.stringify(results, null, 2), 'utf-8');

    // 延迟避免限速
    await new Promise(r => setTimeout(r, 1000));
  }

  console.log(`\n[OK] 翻译完成，保存到: ${OUTPUT_FILE}`);
}

main().catch(console.error);

File: .temp/translate_names.py
"""
批量翻译游戏名称（英文 -> 中文）
使用 Google Translate API (通过 googletrans 库)
"""

import json
import time
import requests
from pathlib import Path

# 输入文件
INPUT_FILE = Path(__file__).parent.parent / "data" / "games_to_translate.json"
OUTPUT_FILE = Path(__file__).parent.parent / "data" / "games_translated.json"

# 加载需要翻译的游戏
with open(INPUT_FILE, "r", encoding="utf-8") as f:
    games = json.load(f)

print(f"需要翻译的游戏数量: {len(games)}")

def translate_text(text, retries=3):
    """使用 LibreTranslate (免费 API) 翻译文本"""
    url = "https://libretranslate.com/translate"
    data = {
        "q": text,
        "source": "en",
        "target": "zh",
        "format": "text"
    }

    for attempt in range(retries):
        try:
            response = requests.post(url, data=data, timeout=10)
            if response.status_code == 200:
                result = response.json()
                return result.get("translatedText", text)
            elif response.status_code == 429:
                print(f"  [!] 限速，等待 5 秒...")
                time.sleep(5)
            else:
                print(f"  [!] 错误: {response.status_code}")
        except Exception as e:
            print(f"  [!] 异常: {e}")

    return text  # 失败时返回原文

def translate_with_my_memory(text):
    """使用 MyMemory 免费翻译 API"""
    url = f"https://api.mymemory.translated.net/get?q={text}&langpair=en|zh"

    try:
        response = requests.get(url, timeout=5)
        if response.status_code == 200:
            result = response.json()
            translated = result.get("responseData", {}).get("translatedText")
            if translated and translated != text:
                return translated
    except:
        pass

    return None

# 翻译所有游戏名称
results = []
batch_size = 100

for i, game in enumerate(games):
    if i % batch_size == 0:
        print(f"\n[进度] {i}/{len(games)}...")

    # 先尝试 MyMemory (更快)
    translated = translate_with_my_memory(game["nameEn"])

    # 如果失败，尝试 LibreTranslate
    if not translated:
        translated = translate_text(game["nameEn"])

    results.append({
        "appId": game["appId"],
        "nameEn": game["nameEn"],
        "nameZh": translated
    })

    # 避免限速
    time.sleep(0.1)

# 保存结果
with open(OUTPUT_FILE, "w", encoding="utf-8") as f:
    json.dump(results, f, ensure_ascii=False, indent=2)

print(f"\n[OK] 翻译完成，保存到: {OUTPUT_FILE}")
